lib/python/qmk/cli/doctor.py

```diff
@@ -154,6 +154,17 @@ def check_submodules():
     return CheckStatus.OK
 
 
+def check_git_repo():
+    """Checks that the .git directory exists inside QMK_HOME.
+
+    This is a decent enough indicator that the qmk_firmware directory is a
+    proper Git repository, rather than a .zip download from GitHub.
+    """
+    dot_git_dir = QMK_FIRMWARE / '.git'
+
+    return CheckStatus.OK if dot_git_dir.exists() else CheckStatus.WARNING
```

Comment:
```suggestion
    return CheckStatus.OK if dot_git_dir.is_dir() else CheckStatus.WARNING
```
If we want to be more pedantic.
This also checks existence.